modules/optimization/src/actions/OptimizationAction.C

```diff
@@ -79,7 +79,7 @@ OptimizationAction::act()
 
     // Set the object parameters
     InputParameters & params = action->getObjectParams();
-    params.set<bool>("kernel_coverage_check") = false;
+    params.set<MooseEnum>("kernel_coverage_check") = false;
```

Comment:
I doubt that this will do the correct thing. This will likely implicitly convert to 0 and consequently correspond to the first of your enumerators (which is `"TRUE"` currently)

Reply:
Use a string instead, e.g. `"false"`

Reply:
and let s reorganize the enum to be fool proof to these implicit conversions

false
true
off
on